protected/yii

```diff
@@ -25,6 +27,11 @@ $config = yii\helpers\ArrayHelper::merge(
     require(__DIR__ . '/config/console.php')
 );
 
-$application = new humhub\components\console\Application($config);
-$exitCode = $application->run();
-exit($exitCode);
+try {
+    $exitCode = (new humhub\components\console\Application($config))->run();
+    exit($exitCode);
+} catch (\Throwable $ex) {
+    if (null === DatabaseHelper::handleConnectionErrors($ex)) {
```

Comment:
@martin-rueegg A small note, maybe it's also worth to include this at CodeStyle docs, but mostly we use the order `$x === null` and not `null === $x`. Maybe something for future PRs :-)

Reply:
I see. Well, it was a slightly different code at some point, then I've simplified it. It was more like a ...

```php
if ($test === $resultToBeUsed = $deep->nested->path->with->magic->access->getValue())  {}
```

... kinda thing. So in this scenario, the `$test`/`null` must be on the left-hand side of the test operator.

Alternatives would be:

```php
$resultToBeUsed = $deep->nested->path->with->magic->access->getValue();
if ($resultToBeUsed === null)  {}
```

or


```php
if (($resultToBeUsed = $deep->nested->path->with->magic->access->getValue()) === null)  {}
```

Both of which are more cluttered (the former) or less readable (the latter), in my perception.

Unfortunately, in humhum I see a lot of this, where the `$resultToBeUsed` is not stored:

```php
if ($deep->nested->path->with->magic->access->value === null)  {
   // do something
} else {
    $deep->nested->path->with->magic->access->value->read();
    $deep->nested->path->with->magic->access->value->write();
    $deep->nested->path->with->magic->access->value->andSomeMore();
}
```
Which is super inefficient, unless you expect that one object in the chain would have been replaced due to the actions taken.

Maybe something to consider in the code style, too. Accessing array entries, particularly if the array has many keys or the value is an array on its own, is expensive.

Also, accessing magic methods is, and the getter function should be used instead (unless it is an un[cached relation](https://github.com/humhub/humhub/blob/9d984e443b486a412247a8986d892885d1344b85/protected/humhub/components/FindInstanceTrait.php#L163) definition).

And such result should be cached within the function or block.

IMO. :-)

Reply:
@martin-rueegg 
> 
> ```
> $resultToBeUsed = $deep->nested->path->with->magic->access->getValue();
> if ($resultToBeUsed === null)  {}
> ```
> 
I personally prefer this one, because it's much more readable.

> Unfortunately, in humhum I see a lot of this, where the `$resultToBeUsed` is not stored:
> 
> ```
> if ($deep->nested->path->with->magic->access->value === null)  {
>    // do something
> } else {
>     $deep->nested->path->with->magic->access->value->read();
>     $deep->nested->path->with->magic->access->value->write();
>     $deep->nested->path->with->magic->access->value->andSomeMore();
> }
> ```

Yes, such deep nesting is really bad, especially in your example the `value` is often used. But depends... 
 
> Maybe something to consider in the code style, too. Accessing array entries, particularly if the array has many keys or the value is an array on its own, is expensive.
> 
> Also, accessing magic methods is, and the getter function should be used instead (unless it is an un[cached relation](https://github.com/humhub/humhub/blob/9d984e443b486a412247a8986d892885d1344b85/protected/humhub/components/FindInstanceTrait.php#L163) definition).
> 
> And such result should be cached within the function or block.
> 
> IMO. :-)

I need to think about the caching topic a bit. I understand that there is really room for improvement here, but I also want to stay as close as possible to the Yii2 framework concepts and also not add too much complexity.